codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/ClientSimpleMethods.java

```diff
@@ -0,0 +1,114 @@
+import software.amazon.awssdk.regions.Region;
+
+public class ClientSimpleMethods implements ClassSpec {
+
+    private static final String APPROVED_VERBS = "(get|list|describe).*";
+
+    private final IntermediateModel model;
+    private final PoetExtensions poetExtensions;
+
+    public ClientSimpleMethods(IntermediateModel model) {
+        this.model = model;
+        this.poetExtensions = new PoetExtensions(model);
+    }
+
+    @Override
+    public TypeSpec poetSpec() {
+
+        ClassName className = poetExtensions.getClientClass("SimpleMethodsIntegrationTest");
+        ClassName interfaceClass = poetExtensions.getClientClass(model.getMetadata().getSyncInterface());
+        List<String> whitelistedApis = model.getCustomizationConfig().getVerifiedSimpleMethods();
+
+        TypeSpec.Builder builder = PoetUtils.createClassBuilder(className)
+                                            .addModifiers(Modifier.PUBLIC)
+                                            .addField(FieldSpec.builder(interfaceClass, "client")
+                                                               .addModifiers(Modifier.PRIVATE, Modifier.STATIC)
+                                                               .build())
+                                            .addMethod(setup());
+
+        model.getOperations().entrySet().stream()
+             .filter(v -> v.getValue().getInputShape().isSimpleMethod())
+             .filter(v -> !whitelistedApis.contains(v.getValue().getMethodName()))
+             .filter(v -> v.getValue().getMethodName().matches(APPROVED_VERBS))
+             .map(o -> simpleMethodsTest(o.getValue()))
+             .forEach(builder::addMethod);
+
+        return builder.build();
+    }
+
+    @Override
+    public ClassName className() {
+        return poetExtensions.getClientClass("SimpleMethodsIntegrationTest");
+    }
+
+    /**
+     * Creates a setup method for instantiating a new client. If no regions are present for a service,
+     * us-east-1 will be used. If the service is available in aws-global, that region will be used. If the
+     * service is not available in aws-global but is in us-east-1, that region will be used. If a service is
+     * not available in us-east-1 or aws-global, the first region in the available regions for a service will
+     * be used.
+     */
+    private MethodSpec setup() {
+        ClassName beforeClass = ClassName.get("org.junit", "BeforeClass");
+        ClassName interfaceClass = poetExtensions.getClientClass(model.getMetadata().getSyncInterface());
+        return MethodSpec.methodBuilder("setup")
+                         .addAnnotation(beforeClass)
+                         .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+                         .beginControlFlow("if ($T.serviceMetadata().regions().isEmpty())", interfaceClass)
+                         .addStatement("client = $T.builder().region(Region.US_EAST_1).build()", interfaceClass)
```

Comment:
I don't think performance is *that* critical during generation. I think we should use placeholders in the majority of places to protect ourselves from future refactors.